95/23 Add San Diego Experiments and XC downloader

Sean1572 opened the pull request.

Sean1572 committed:
Adds working code for XC data and few shot model training

Sean1572 committed:
Add soundfile

Sean1572 committed:
Add code for using XC api

Sean1572 committed:
Bird mae (#94)

* Added some experimental inferance pipeline

Needed this to test unseen burrowing owl dataset. Will help alot with getting data formatted correctly

* Add testing code for loading in models for inferance

* feat: add inferance pipeline

* Block CSVs from commits

* Move visualization packages to optional dependecy

* Clean code (remove comments and remove run_spefific code from library)

* Clean up timm_model

* Add BirdMAE Model Training

Sean1572 committed:
Reslove train.py conflicts

Sean1572 committed:
Fix bugs with birdmae data loading

Sean1572 commented:
Merge only after merging inference pipeline see #88

Sean1572 committed:
Linted (round 1)

Sean1572 committed:
Remove perch model

Sean1572 committed:
fix: adjust for nas data

Sean1572 committed:
lint: finsh flake8 linting

Sean1572 committed:
Clean up few-shot experiment branch

Sean1572 committed:
Merge branch 'dev' into few_shot_experiments

Sean1572 committed:
Add inferance fixes for fewshot

Sean1572 committed:
Add fixes to model training

BUG NOTICE: Continue broke inferance in waveform_preprocessors.py... need to rethink how addressing corrutpted audio files works in model training. Maybe replace with empty data so it doesn't break training?

kgarwoodsdzwa commented:
@Sean1572 was thinking of merging this before doing the push to main, but there's merge conflicts with the dev branch. i believe mainly with the inference script, train script, and trainer script, and raw data extractor etc. maybe this could be resolved by putting all the revised ones needed in a jupyter notebook in an example folder?

Sean1572 committed:
Merge branch 'dev' into few_shot_experiments

Sean1572 commented on whoot_model_training/whoot_model_training/preprocessors/waveform_preprocessors.py:
```diff
@@ -0,0 +1,147 @@
+            except Exception as e:
+                print(e)
+                print("File Likely is corrupted, moving on")
+                break
```
@kgarwoodsdzwa I realized an issue with this section of the code, part of the reason I didn't want it pushed to main

This was my cheap way of handling file corruption. Previously this was a continue and it was fine (just a smaller batch size). This however is a huge issue in inference because it can disalign what is the file name to the model output (I passed in 15 files, but only get 14 predictions...). Raising an error fixed this because I could just skip the batch. During training we have no such error handling since trainer doesn't allow for it.

So we need to tackle error handling as part of this repo

Sean1572 committed:
Fix bug with corrupted files

Sean1572 committed:
Outdated inference script

Sean1572 committed:
Lint round 1

Linted most the model_trainer with flake8 and some of data downloader, realized that data downloader needs a major code clean up

Sean1572 committed:
Linited Data Downloader

Sean1572 committed:
Linted but needs testing

Sean1572 committed:
Fixed bugs from linting so it works with Buowset v1.3

kgarwoodsdzwa committed:
Merge branch 'dev' into few_shot_experiments
